charts/fluid/fluid/templates/controller/alluxioruntime_controller.yaml

```diff
@@ -3,7 +3,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: alluxioruntime-controller
-  namespace: fluid-system
+  namespace: {{ .Release.Namespace }}
```

Comment:
This is a general approach, such as the following project
https://github.com/kubevela/kubevela/blob/master/charts/vela-core/templates/kubevela-controller.yaml#L200
https://github.com/crossplane/crossplane/blob/master/cluster/charts/crossplane/templates/deployment.yaml#L5

But given the compatibility of previous strategy

```yaml
# values.yaml
# If the user wants to use -n mode, remove the default value of values
namespace: "fluid-system"
```
```tpl
# _helpers.tpl
{{- define "fluid.namespace" -}}
{{- if .Values.namespace }}
{{- .Values.namespace  }}
{{- else }}
{{- .Release.Namespace -}}
{{- end }}
{{- end }}
```
```yaml
kind: Deployment
metadata:
  name: alluxioruntime-controller
  namespace:  {{ include "fluid.namespace" . }}
```

If the user wants to use -n mode, remove the default value of values

Reply:
Use --set to override the default 
helm install fluid  ./fluid -n test --set namespace=''